lineapy/graph_reader/artifact_collection.py

```diff
@@ -118,65 +140,49 @@ def _sort_session_artifacts(
         support such circular dependencies between sessions,
         which is a future project.
         """
-        # Construct a combined graph across multiple sessions
-        combined_graph = nx.DiGraph()
-        for session_artifacts in self.session_artifacts.values():
-            session_graph = session_artifacts.graph.nx_graph
-            combined_graph.add_nodes_from(session_graph.nodes)
-            combined_graph.add_edges_from(session_graph.edges)
-
-        # Check if user-specified dependency includes artifacts not in the current collection
-        dependency_edges = list(
+
+        # Merge task graph from all sessions
+        combined_taskgraph = nx.union_all(
+            [
+                sa.nodecollection_dependencies.graph
+                for session_id, sa in self.session_artifacts.items()
+            ]
+        )
+        # Add edge for user specified dependencies
+        task_dependency_edges = list(
             chain.from_iterable(
-                ((artname, to_artname) for artname in from_artname)
+                (
+                    (artname.replace(" ", "_"), to_artname.replace(" ", "_"))
+                    for artname in from_artname
+                )
                 for to_artname, from_artname in dependencies.items()
             )
         )
-        dependency_nodes = set(chain(*dependency_edges))
-        missing_artifacts = [
-            artname
-            for artname in dependency_nodes
-            if self.art_name_to_node_id.get(artname, None) is None
-        ]
-        if len(missing_artifacts) > 0:
-            raise KeyError(
-                f"Dependency graph includes artifact(s) not in this artifact collection: {missing_artifacts}"
-            )
-
-        # Augment the graph with user-specified edges
-        dependency_edges_by_id = [
-            (
-                self.art_name_to_node_id.get(edge_tuple[0]),
-                self.art_name_to_node_id.get(edge_tuple[1]),
-            )
-            for edge_tuple in dependency_edges
-        ]
-        combined_graph.add_edges_from(dependency_edges_by_id)
-
+        task_dependency_nodes = set(chain(*task_dependency_edges))
+        for artname in task_dependency_nodes:
+            if artname not in list(combined_taskgraph.nodes):
+                raise KeyError(
```

Comment:
Prefer the old behavior here that gives a list of all missing artifacts instead of just the first one.